packages/dds/tree/src/test/simple-tree/benchmarkUtilities.ts

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import {
+	SchemaFactory,
+	type NodeFromSchema,
+	type Unhydrated,
+	type ValidateRecursiveSchema,
+} from "../../simple-tree/index.js";
+import { hydrate } from "./utils.js";
+import { Tree } from "../../shared-tree/index.js";
+
+const schemaFactory = new SchemaFactory("test");
+
+/**
+ * A recursive tree node with a single leaf value
+ */
+export class DeepTreeNode extends schemaFactory.objectRecursive("deep-tree-node", {
+	node: [() => DeepTreeNode, schemaFactory.number],
+}) {}
+{
+	type _check = ValidateRecursiveSchema<typeof DeepTreeNode>;
+}
+
+/**
+ * A wide tree node with all leaves having the same value
+ */
+export const WideTreeNode = schemaFactory.array(schemaFactory.number);
+export type WideTreeNode = NodeFromSchema<typeof WideTreeNode>;
+
+/**
+ * Make an unhydrated deep tree with a single leaf value
+ */
+function makeDeepTree(depth: number, leafValue: number): Unhydrated<DeepTreeNode> {
+	return new DeepTreeNode({ node: depth === 1 ? leafValue : makeDeepTree(depth - 1, leafValue) });
+}
+
+/**
+ * generate a deep tree with a single leaf value
+ */
+export function generateDeepSimpleTree(depth: number, leafValue: number): DeepTreeNode {
+	assert(Number.isSafeInteger(depth));
+	assert(depth > 0);
+
+	return hydrate(DeepTreeNode, makeDeepTree(depth, leafValue));
+}
+
+/**
+ * read the deep tree and return the depth and value of the leaf node
+ */
+export function readDeepSimpleTree(tree: DeepTreeNode): {
+	depth: number;
+	value: number;
+} {
+	let currentNode: DeepTreeNode | number = tree;
+	let depth = 0;
+
+	while (Tree.is(currentNode, DeepTreeNode)) {
+		depth += 1;
+		currentNode = currentNode.node;
+	}
+
+	return { depth, value: currentNode };
+}
+
+/**
+ * Update the deep tree with a new leaf value deep in the tree
+ */
+export function writeDeepSimpleTree(tree: DeepTreeNode, newValue: number): void {
```

Comment:
Since these functions are only used in a simple tree specific context, I think the names can be simplified to remove "simple" from them.